apps/analysis/serializers.py

```diff
@@ -459,9 +460,13 @@ class Meta:
     def validate_analysis_pillar(self, analysis_pillar):
         if analysis_pillar.analysis.project != self.context['request'].active_project:
             raise serializers.ValidationError('Invalid analysis pillar')
+        if self.context['request'].active_project.is_private:
+            raise serializers.ValidationError('Topic model is not allowed for private projects')
         return analysis_pillar
 
     def validate_additional_filters(self, additional_filters):
+        # NOTE: overlapping the lead_confidentialities filter, only public leads are allowed
+        additional_filters['lead_confidentialities'] = [Lead.Confidentiality.UNPROTECTED]
```

Comment:
Let's move this logic to 


apps/analysis/models.py:473
```python3
    def get_entries_qs(self):
        additional_filters = copy.deepcopy(self.additional_filters)
        # NOTE: NLP is using LLM, to avoid data leakage we only pass UNPROTECTED
        additional_filters['lead_confidentialities'] = [Lead.Confidentiality.UNPROTECTED]
        return self._get_entries_qs(self.analysis_pillar, additional_filters)
```